informer/spec.go

```diff
@@ -0,0 +1,19 @@
+package informer
+
+import (
+	"k8s.io/client-go/pkg/runtime"
+)
+
+// Observer functions are called when the cache.ListWatch List and Watch
+// functions are called.
+type Observer interface {
+	OnList()
+	OnWatch()
+}
+
+// ZeroObjectFuncs provides zero values of an object and objects' list ready to
+// be decoded. The provided zero values must not be reused by zeroObjectFactory.
+type ZeroObjectFactory interface {
+	NewObject() runtime.Object
+	NewObjectList() runtime.Object
+}
```

Comment:
We tend to put interface definitions in `spec.go`. I think this works. You directly know where to find interfaces of a package and which ones are there and how many, because they are all in one place. I really dislike having things spread and hidden in different files and at the top or at the bottom and then it is all a mess.